cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeServer.java

```diff
@@ -86,6 +84,16 @@ public void modify(ChannelPipeline pipeline) {
     this.discoveryService = discoveryService;
   }
 
+  /**
+   * Enable the runtime identity handler if security is enabled AND either internal auth is enabled OR runtime identity
+   * backwards compatibility mode is disabled.
+   */
+  private boolean enableRuntimeIdentity(CConfiguration cConf) {
+    return cConf.getBoolean(Constants.Security.ENABLED, false)
+      && SecurityUtil.isInternalAuthEnabled(cConf)
```

Comment:
I believe && has higher precedence, so this mean A && B || C equivalent to (A && B) || C, see https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html

Reply:
Fixed, thanks!